plugins/inputs/cisco_telemetry_gnmi/cisco_telemetry_gnmi.go

```diff
@@ -324,23 +323,9 @@ func (c *CiscoTelemetryGNMI) handleTelemetryField(update *gnmi.Update, tags map[
 		return aliasPath, nil
 	}
 
+	value, _ := gnmivalue.ToScalar(update.Val)
```

Comment:
Check the error and also there are some value types we should probably skip, such as []byte.  The field needs to be one of int64, float64, uint64, string, bool, there is some massaging that is done when acc.AddField is called but that was done mostly to avoid errors with older plugins, we try do it in the plugin now.

Is this going to work with the JsonVal/JsonIetfVal types?  Maybe we should do the ToScaler as the default case of the switch?

Reply:
I see your points, I therefore reverted those changes and just changed the handling for the decimalval type. I think in hindsight it would be more trouble switching to the native casting function since we would then need to deal with the error cases as it also does not deal with the Json types which we would need to handle manually anyway.